code/game/objects/items/toys.dm

```diff
@@ -603,7 +603,12 @@
 
 
 /obj/item/toy/plushie/proc/explosive_betrayal(obj/item/grenade/grenade_callback)
+	var/grenade_inside = FALSE //Any grenade, even non-explosive, will destroy the plushie.
+	if(grenade)
+		grenade_inside = TRUE
 	grenade_callback.prime()
+	if(grenade_inside && !QDELETED(src))
+		qdel(src)
```

Comment:
Mention these changes in the PR description and changelog

Reply:
Also, is it possible this might delete the grenade before the explosion?

Reply:
Done